gc.c

```diff
@@ -2505,6 +2530,14 @@ newobj_init(VALUE klass, VALUE flags, int wb_protected, rb_objspace_t *objspace,
     p->as.basic.flags = flags;
     *((VALUE *)&p->as.basic.klass) = klass;
 
+    int t = flags & RUBY_T_MASK;
+    if (t == T_CLASS || t == T_MODULE || t == T_ICLASS) {
+        RVALUE_AGE_SET_CANDIDATE(objspace, obj);
+    }
+    else {
+        RVALUE_AGE_SET(obj, 0);
+    }
+
```

Comment:
Do we need this else clause? The age should have been set to 0 when we put it on the freelist right?